animation_nodes/nodes/gpencil/gp_layer_info.py

```diff
@@ -0,0 +1,150 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... data_structures import GPFrame, LongList
+from ... base_types import AnimationNode, VectorizedSocket
+
+frameTypeItems = [
+    ("ACTIVE", "Active Frame", "Get active Frame", "NONE", 0),
+    ("INDEX", "Frame Index", "Get specific Frame with index", "NONE", 1),
+    ("FRAME", "Frame Number", "Get specific Frame with frame-number", "NONE", 2),
+    ("ALL", "All Frames", "Get all frames", "NONE", 3)
+]
+
+class GPLayerInfoNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_GPLayerInfoNode"
+    bl_label = "GP Layer Info"
+    errorHandlingType = "EXCEPTION"
+
+    frameType: EnumProperty(name = "Frame Type", default = "ACTIVE",
+        items = frameTypeItems, update = AnimationNode.refresh)
+
+    useFrameList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("GPLayer", "Layer", "layer", dataIsModified = True)
+
+        if self.frameType == "ACTIVE":
+            self.newInput("Scene", "Scene", "scene", hide = True)
+            self.newOutput("GPFrame", "Frame", "frame")
+            self.newOutput("Float", "Frame Number", "frameNumber")
```

Comment:
I would say the Frame Number doesn't belong here. If the user wants to get it, it can be computed using the Frame Info node.

Reply:
Should I remove Frame Number from all the cases?

Reply:
Yes.

Reply:
Okay

Reply:
Removed.